lib/foreman_ansible_core/runner/ansible_runner.rb

```diff
@@ -106,6 +108,22 @@ def write_playbook
         File.write(File.join(@root, 'project', 'playbook.yml'), @playbook)
       end
 
+      def write_ssh_key
+        key_path = File.join(@root, 'env', 'ssh_key')
+        File.symlink(ForemanRemoteExecutionCore.settings[:ssh_identity_key_file], key_path)
+
+        if !@passphrase.nil? && !@passphrase.empty?
+          passwords_path = File.join(@root, 'env', 'passwords')
+          secrets = <<~SECRETS
+            ---
+            "for.*/artifacts/.*/ssh_key_data:": "#{@passphrase}"
```

Comment:
Is this supposed to be yaml? Shouldn't we build a hash and `YAML.dump` it to avoid escaping issues?